gcp_variant_transforms/transforms/limit_write.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2017 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A PTransform for generating some random keys and group input elements (whcih
+are BigQuery rows) based on the dummy key before writing them to output table.
+This is a work around to avoid the following failure:
+    BigQuery execution failed., Error:
+    Message: Too many sources provided: xxxxx. Limit is 10000.
+
+"""
+
+
+from __future__ import absolute_import
+
+import apache_beam as beam
+import random
+
+__all__ = ['LimitWrite']
+
+
+class _RoundRobinKeyFn(beam.DoFn):
+  def __init__(self, count):
+    self.count = count
+
+  def start_bundle(self):
```

Comment:
This is one of the DoFn methods, it will be called before a worker starts to process a new bundle.

https://github.com/apache/beam/blob/e30e0c807321934e862358e1e3be32dc74374aeb/sdks/python/apache_beam/transforms/core.py#L329

Reply:
Ah, cool!